middlewares/authenticate.js

```diff
@@ -35,7 +56,7 @@ module.exports = async (req, res, next) => {
     const userData = await users.fetchUser({ userId })
     req.userData = userData.user
 
-    return next()
+    return checkRestricted(req, res, next)
```

Comment:
IMO, the authenticate function shouldn't be checking the restricted users, it should just check the auth. 🤔
Is there any way, we can add the `checkRestricted` as a global middleware, so that it will apply to every request?

Reply:
No. It cannot be done that way because to check for `restricted` this middleware needs to come after the authenticate middleware.

Currently we add `authenticate` middleware explicitly while adding any new route. So this middleware would have to be added explicitly after each location where `authenticate` is used.

To avoid this repetition, I have called this middleware from within the authenticate middleware itself.